src/compiler/scanner.ts

```diff
@@ -598,6 +601,42 @@ module ts {
             ch >= CharacterCodes._0 && ch <= CharacterCodes._9 || ch === CharacterCodes.$ || ch === CharacterCodes._ ||
             ch > CharacterCodes.maxAsciiCharacter && isUnicodeIdentifierPart(ch, languageVersion);
     }
+    
+    /* @internal */ 
+    export function retrieveClosingTagsMap(text: string, languageVersion: ScriptTarget): Map<number[]> {
+        let pos = 0;
+        let end = text.length;
+        const closingTags: Map<number[]> = {};
+        while (pos < end) {
+            if (text.charCodeAt(pos) === CharacterCodes.lessThan && text.charCodeAt(pos + 1) === CharacterCodes.slash) {
+                let p = pos + 2;
+                let ch: number;
+                while (p < end) {
+                    ch = text.charCodeAt(p);
+                    if (!isIdentifierPart(ch, languageVersion) &&
+                        ch !== CharacterCodes.minus &&
+                        ch !== CharacterCodes.dot &&
+                        ch !== CharacterCodes.backslash &&
+                        !isWhiteSpace(ch) &&
+                        !isLineBreak(ch)) {
+                        break;
+                    }
+                    p++;
+                }
+                if (ch === CharacterCodes.greaterThan) {
+                    let tagName = text.substring(pos + 2, p).replace(/\s/g, '');
+                    if (!hasProperty(closingTags, tagName)) {
+                        closingTags[tagName] = [];
+                    }
+                    closingTags[tagName].push(pos);
+                    pos = p + 1;

```

Comment:
I think it should be `pos = p;` because you are about to increment pos at the end of the loop iteration. Seems like this would break the case where you have two closing tags in a row with no space in between.